Sources/Rendering/OpenGL/Texture/index.js

```diff
@@ -627,11 +716,17 @@ function vtkOpenGLTexture(publicAPI, model) {
     ) {
       for (let idx = 0; idx < data.length; idx++) {
         if (data[idx]) {
-          const dataArrayToCopy =
-            data[idx].length > pixCount
-              ? data[idx].subarray(0, pixCount)
-              : data[idx];
-          pixData.push(new Float32Array(dataArrayToCopy));
+          if (updateRegions) {
+            pixData.push(
+              new Float32Array(readRegions(data[idx], volumeRegions))
+            );
+          } else {
+            const dataArrayToCopy =
+              data[idx].length > pixCount
+                ? data[idx].subarray(0, pixCount)
+                : data[idx];
+            pixData.push(new Float32Array(dataArrayToCopy));
```

Comment:
You can move the effect of `constructor(subarray())`  into `constructor(data, offset, length)`, you avoid the creation of a sub array

Reply:
fair enough